### ISSUE TITLE
Text in embedded editor does not look good

### ISSUE THREAD

**Issue by kupiqu** [flathub/org.octave.Octave#13]
It appears just fine elsewhere:

![image](https://user-images.githubusercontent.com/3875196/38159937-b7cdafa8-3481-11e8-876b-37e016111206.png)

This is most likely related to KDE scaling, but note that this doesn't happen in Octave 4.2.1, so I guess there is sth missing in Octave or KDE flatpak packages

**Comment by Alexander-Wilms** [comment]
Hm, cannot reproduce this on KDE. Do you have a hidpi screen?

Maybe related: https://bitbucket.org/tortoisehg/thg/issues/5010/good-and-poor-display-within-the-same-thg

**Comment by kupiqu** [comment]
> Hm, cannot reproduce this on KDE. Do you have a hidpi screen?
>
Yes, hidpi with kde scaling

> Maybe related:
> https://bitbucket.org/tortoisehg/thg/issues/5010/good-and-poor-display-within-the-same-thg
>
I think this is related indeed. Would qscintilla need to be updated to fix
the issue? Is this embedded within Octave, right?

**Comment by kupiqu** [comment]
I see, thanks. I'll check recent qsicintilla commits to see if there is anything related to this.

**Comment by kupiqu** [comment]
couldn't find the source repo for qscintilla :/

what I don't understand yet is why octave 4.2.1 doesn't suffer the problem. wondering wheter it may be a qscintilla regression of some sort?

I'll try to compile octave

**Comment by mtmiller** [comment]
4.2.1 from the Ubuntu 17.10 repos? It is built against Qt 4, not quite apples to apples.

**Comment by kupiqu** [comment]
I see. That makes sense

**Comment by Alexander-Wilms** [comment]
Here's the QScintilla sources: https://www.riverbankcomputing.com/software/qscintilla/download

**Comment by kupiqu** [comment]
> Here's the QScintilla sources: https://www.riverbankcomputing.com/software/qscintilla/download

Thanks!

I'm running a bit busy right now, but testing this is in my agenda...

**Comment by kupiqu** [comment]
I'm trying to comile Octave to test this, but I get this error on bootstrap:

```
❯ ./bootstrap 
./bootstrap: Error: --no-git requires --gnulib-srcdir
```

even though git is set to true in the file :/

**Comment by mtmiller** [comment]
I have a patch that may or may not improve things with QScintilla on a HiDPI display, need someone to build and test it. It has no effect on my normal display.

**Comment by kupiqu** [comment]
> I have a patch that may or may not improve things with QScintilla on a HiDPI display, need someone to build and test it. It has no effect on my normal display.

Thanks, Mike!

I'm very interested in testing it but I'm having troubles building Octave (please see comment above). Would it require a lot of effort from you to create a test flatpak package? Unless I find a way to fix the bootstrap error, I think this is probably the easiest way to try the patch out...

**Comment by Alexander-Wilms** [comment]
You don't need to build Octave separately any more, since there are now official tarballs. Just run flatpak-builder and it should just work.

**Comment by kupiqu** [comment]
> You don't need to build Octave separately any more, since there are now official tarballs. Just run flatpak-builder and it should just work.

But what about the patch, I apply it to the tarball?

**Comment by mtmiller** [comment]
Please try building from the qscintilla-hidpi branch.

**Comment by mtmiller** [comment]
It's a git submodule ref, you'll have to `git submodule update --init`.

You should add `--repo=repo` to the flatpak-builder command line to automatically export the app to a local repo directory after a successful build.

**Comment by kupiqu** [comment]
Now it seems it's doing sth :)

**Comment by kupiqu** [comment]
Now, I ge the following error:

```
Initializing build dir
error: Requested extension org.freedesktop.Sdk.Extension.gfortran-62 not installed
Error: Child process exited with code 1
```

**Comment by Alexander-Wilms** [comment]
Run `flatpak install flathub org.freedesktop.Sdk.Extension.gfortran-62 org.freedesktop.Sdk.Extension.openjdk9`

**Comment by mtmiller** [comment]
Yeah, more setup :man_shrugging: You need to manually install the two "Extension" runtimes mentioned in the manifest file.

**Comment by kupiqu** [comment]
Thank you! Getting closer to get it done... :)

**Comment by kupiqu** [comment]
Good news. The patch works. No more blurry text in the editor :)

**Comment by mtmiller** [comment]
Great! I will try to get this into the release after some more testing on other platforms.

**Comment by kupiqu** [comment]
Thank you!

**Comment by Alexander-Wilms** [comment]
Does the patch still need more testing?

**Comment by mtmiller** [comment]
The patch has been tested and found to not be necessary and have no ill effects on non-GNU/Linux systems. I have no problem with merging this into the 4.4.0 flatpak build. I do expect to include it in 4.4.1.

**Comment by kupiqu** [comment]
> I have no problem with merging this into the 4.4.0 flatpak build

That would be very much appreciated

**Comment by kupiqu** [comment]
@Alexander-Wilms, would you consider merging the patch into current flatpak build?

@mtmiller, do you know if there is a time frame for the release of Octave 4.4.1?

Thank you both!

**Comment by mtmiller** [comment]
No I don't know when 4.4.1 will be released.

**Comment by mtmiller** [comment]
This patch is now included in the Octave stable branch for the 4.4.1 release.

https://hg.savannah.gnu.org/hgweb/octave/rev/5daeb4049088

**Comment by kupiqu** [comment]
Great, looking forward...
